tests/std/tests/P0355R7_calendars_and_time_zones_dates/test.cpp

```diff
@@ -0,0 +1,930 @@
+    } else {
+        for (auto m = 0u; m <= 255u; ++m) {
+            for (auto wd = 0u; wd <= 255u; ++wd) {
+                for (auto wdi = 0u; wdi <= 6u; ++wdi) {
+                    const auto mwd = month{m} / weekday{wd}[wdi];
+                    if (m >= 1 && m <= 12 && wd <= 7 && wdi >= 1 && wdi <= 5) {
+                        assert(mwd.ok());
+                    } else {
+                        assert(!mwd.ok());
+                    }
+                }
+            }
+        }
+    }
+}
+
+constexpr void month_weekday_last_test() {
+    const auto mwdl = January / Monday[last];
+
+    static_assert(noexcept(month_weekday_last{month{}, weekday_last{Sunday}}));
+    static_assert(noexcept(mwdl.month()));
+    static_assert(noexcept(mwdl.weekday_last()));
+    static_assert(noexcept(mwdl.ok()));
+
+    static_assert(noexcept(mwdl == mwdl));
+
+    assert(mwdl.month() == January);
+    assert(mwdl.weekday_last().weekday() == Monday);
+    assert(mwdl == January / Monday[last]);
+}
+
+constexpr void year_month_test() {
+    auto ym = 2020y / January;
+
+    static_assert(noexcept(year_month{}));
+    static_assert(noexcept(year_month{year{}, month{}}));
+
+    static_assert(noexcept(ym.year()));
+    static_assert(noexcept(ym.month()));
+
+    static_assert(noexcept(ym += months{}));
+    static_assert(noexcept(ym -= months{}));
+    static_assert(noexcept(ym += years{}));
+    static_assert(noexcept(ym -= years{}));
+
+    static_assert(noexcept(ym.ok()));
+
+    static_assert(noexcept(ym == ym));
+    static_assert(noexcept(ym <=> ym));
+
+    static_assert(noexcept(ym + months{}));
+    static_assert(noexcept(months{} + ym));
+    static_assert(noexcept(ym - months{}));
+
+    static_assert(noexcept(ym - ym));
+    static_assert(noexcept(ym + years{}));
+    static_assert(noexcept(years{} + ym));
+    static_assert(noexcept(ym - years{}));
+
+    assert(ym.year() == 2020y);
+    assert(ym.month() == January);
+
+    ym += months{2};
+    assert(ym.year() == 2020y);
+    assert(ym.month() == March);
+    ym -= months{2};
+    assert(ym.year() == 2020y);
+    assert(ym.month() == January);
+
+    ym += years{2};
+    assert(ym.year() == 2022y);
+    assert(ym.month() == January);
+    ym -= years{2};
+    assert(ym.year() == 2020y);
+    assert(ym.month() == January);
+
+    assert(2020y / April == 2020y / April);
+    assert(2019y / April < 2020y / April);
+    assert(2020y / March < 2020y / April);
+    assert(2020y / April > 2019y / April);
+    assert(2020y / April > 2020y / March);
+
+    assert(ym + months{2} == 2020y / March);
+    assert(months{2} + ym == 2020y / March);
+
+    assert(ym - months{2} == 2019y / November);
+    assert(ym - 2019y / January == months{12});
+
+    assert(ym + years{2} == 2022y / January);
+    assert(years{2} + ym == 2022y / January);
+
+    assert(ym - years{2} == 2018y / January);
+
+    if (is_constant_evaluated()) {
+        static_assert((2020y / 01).ok());

```

Comment:
`01` is an octal literal. I don't think it's a good idea to write `01` even if `01 == 1`.

Same below (~20 occurrences).